zookeeper-server/src/main/java/org/apache/zookeeper/common/SSLUtil.java

```diff
@@ -0,0 +1,105 @@
+ * limitations under the License.
+ */
+
+package org.apache.zookeeper.common;
+
+import java.lang.reflect.InvocationTargetException;
+import org.apache.zookeeper.common.crypto.Crypt;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SSLUtil {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SSLUtil.class);
+    private static Crypt crypt;
+
+    /**
+     * This method will return an instance of the user configured Crypto class
+     * value for {@value ZKConfig#CONFIG_CRYPT_CLASS}.
+     *
+     * @return an instance of the Crypto class configured
+     */
+    private static Crypt getCrypt() {
+        if (crypt == null) {
+            String cryptClassName = System
+                    .getProperty(ZKConfig.CONFIG_CRYPT_CLASS);
+            Class<?> cryptClass = null;
+            if (cryptClassName != null && !cryptClassName.isEmpty()) {
+                try {
+                    cryptClass = Class.forName(cryptClassName.trim());
+                } catch (ClassNotFoundException e) {
+                    throw new RuntimeException("Class configured for "
+                            + ZKConfig.CONFIG_CRYPT_CLASS + " is not found ", e);
+                }
+                try {
+                    crypt = (Crypt) cryptClass.getConstructor().newInstance();
+                } catch (NoSuchMethodException | InstantiationException
+                        | IllegalAccessException | InvocationTargetException e) {
+                    throw new RuntimeException(
+                            "Could not access default constructor for the Class which is configured for "
+                                    + ZKConfig.CONFIG_CRYPT_CLASS
+                                    + " property ", e);
+                }
+
+            } else {
+                throw new RuntimeException(
+                        "Class to decrypt the encrypted text is not configured for "
+                                + ZKConfig.CONFIG_CRYPT_CLASS + " property ");
+            }
+        }
+        return crypt;
+    }
+
+    /**
+     * This method will decrypt the given text using the Crypto class client
+     * has configured if client has set the password encryption to true
+     *
+     * @param pwd     string to decrypt
+     * @param decrypt client configured value whether the text is in plain or
+     *                encrypted format
+     * @return decrypted text
+     */
+    public static String getDecryptedText(String pwd, boolean decrypt) {
+        if (decrypt) {
+            try {
+                getCrypt();
+            } catch (Exception e) {
+                throw new RuntimeException(
+                        "Failed to get the Crypt reference used for decrypting a text.",
+                        e);
+            }
+            try {
+                pwd = crypt.decrypt(pwd);
+            } catch (Exception e) {
+                throw new RuntimeException(
```

Comment:
Don't swallow the original exception. There won't be any info in the logs about what happened originally.

Reply:
I don't see benefit in catching exception for each call. I think you could just remove the catch blocks altogether, or create only a single one at the end of this method.